rubicon_ml/client/config.py

```diff
@@ -38,12 +38,24 @@ class Config:
     def __init__(
         self, persistence=None, root_dir=None, is_auto_git_enabled=False, **storage_options
     ):
-        self.persistence, self.root_dir, self.is_auto_git_enabled = self._load_config(
-            persistence, root_dir, is_auto_git_enabled
-        )
-        self.storage_options = storage_options
+        if storage_options is not None and storage_options.get("composite_config") is not None:
+            composite_config = storage_options.get("composite_config")
+            self.storage_options = storage_options
+            repositories = []
+            for config in composite_config:
+                self.persistence, self.root_dir, self.is_auto_git_enabled = self._load_config(
```

Comment:
are `self.persistence` and `self.root_dir` accessed at all outside of `self._get_repository`? it seems like `self.persistence` is used to choose the repo then `self.root_dir` gets set on the repo then is always accessed as `repo.root_dir` afterwards. if so we should refactor later

Reply:
`self._get_repository()` calls `self._get_protocol` so they're getting used there again. Seems like the factoring is okay